pkg/frontend/querymiddleware/codec.go

```diff
@@ -427,6 +443,22 @@ func (c prometheusCodec) EncodeResponse(ctx context.Context, res Response) (*htt
 	return &resp, nil
 }
 
+func (prometheusCodec) negotiateContentType(acceptHeader string) (string, formatter) {
+	if acceptHeader == "" {
+		return jsonMimeType, defaultFormatter
```

Comment:
I needed an alternative name to avoid naming collisions with the `jsonFormatter` type. I've renamed `defaultFormatter` to `jsonFormatterInstance` to make it clearer in 712a9a1.